hibernate-reactive-core/src/main/java/org/hibernate/reactive/event/impl/DefaultReactiveInitializeCollectionEventListener.java

```diff
@@ -81,7 +81,8 @@ public void onInitializeCollection(InitializeCollectionEvent event) throws Hiber
 						} );
 			}
 		}
-		throw new AssertionError("Shouldn't be here");
+		// Collection was already initialized.
+		return CompletionStages.nullFuture();
```

Comment:
Haha, seriously!? That's all it was?

OMG, I was right there looking at that line myself.

Reply:
Yes, I think that's it. :)

The tricky bit was that there were multiple non-lazy collections to initialize. When the first one got initialized, the second one also got initialized, so it was the second one that caused the `AssertionError`.

Reply:
Well, good :-)

Thanks